File: AspNetMvcUnleashed/0672329980_source/Chapter11/CS/MvcApplication1/MvcApplication1/Controllers/MovieRepositoryController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using System.Web.Mvc.Ajax;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class MovieRepositoryController : Controller
    {

        private MovieRepositoryBase _repository;

        public MovieRepositoryController() 
            : this(new MovieRepository()) { }

        public MovieRepositoryController(MovieRepositoryBase repository)
        {
            _repository = repository;
        }


        public ActionResult Index()
        {
            return View(_repository.ListMoviesCached());
        }


        public ActionResult Create()
        {
            return View();
        }

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create([Bind(Exclude="Id")]Movie movieToCreate)
        {
            if (!ModelState.IsValid)
                return View();
            _repository.CreateMovie(movieToCreate);
            return RedirectToAction("Index");
        }


    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter11/VB/MvcApplication1/MvcApplication1/Controllers/MovieRepositoryController.vb
Public Class MovieRepositoryController
    Inherits Controller

    Private _repository As MovieRepositoryBase

    Public Sub New()
        Me.New(New MovieRepository())
    End Sub

    Public Sub New(ByVal repository As MovieRepositoryBase)
        _repository = repository
    End Sub


    Public Function Index() As ActionResult
        Return View(_repository.ListMoviesCached())
    End Function


    Public Function Create() As ActionResult
        Return View()
    End Function

    <AcceptVerbs(HttpVerbs.Post)> _
    Public Function Create(<Bind(Exclude:="Id")> ByVal movieToCreate As Movie) As ActionResult
        If (Not ModelState.IsValid) Then
            Return View()
        End If
        _repository.CreateMovie(movieToCreate)
        Return RedirectToAction("Index")
    End Function


End Class
